recipes-core/initrdscripts/files/init-restore-mode.sh

```diff
@@ -106,50 +106,59 @@ mount_nirecovery_usb()
 
 mount_payload()
 {
-   if [ -d $NIRECOVERY_MOUNTPOINT/payload ]; then
-      # We are booting from USB so payload will be on same partition as kernel and initrd
-      mount --bind $NIRECOVERY_MOUNTPOINT/payload /payload;
-   else
-      # We used migration ipk, so payload would be in nirootfs
-      ROOTFS_MOUNTPOINT=$(mktemp -d "/var/volatile/rootfs-XXXXXXX")
-
-      if [ -z  $ROOTFS_MOUNTPOINT ]
-      then
-         echo ""
-         echo "ERROR: Failed to create temporary directory to mount nirootfs"
-         echo ""
-
-         sync
-         show_console
-
-         exit 1
-      fi
-
-      mount -o rw -L nirootfs $ROOTFS_MOUNTPOINT;
-
-      # Delete everything except payload directory to ensure device
-      # has enough space to extract payload
-      for dir in $ROOTFS_MOUNTPOINT/*; do
-          if [ ${dir##*/} != "payload" ]; then
-              rm -rf $dir;
-          fi
-      done
-
-      # Payload is in an ipk, extract it
-      mkdir -p $ROOTFS_MOUNTPOINT/payload/extracted
-      pushd $ROOTFS_MOUNTPOINT/payload/extracted > /dev/null
-
-      ar x $ROOTFS_MOUNTPOINT/payload/*.ipk
-      mkdir data
-      tar -xpf data.tar* -C data
-
-      # Ramdisk size is limited, so copy payload to a tmpfs
-      mount -t tmpfs -o size=420m tmpfs /payload
-      cp data/usr/share/nilrt/* /payload
-      mv /payload/*.raucb /payload/niboot.raucb
-      popd > /dev/null
-
-      umount $ROOTFS_MOUNTPOINT;
+   # Locate and mount the payload at /payload (in the case of initrd having the payload, do nothing)
+   # 1) Check in the current directory (ie built into the initrd as with rauc->safemode migration)
+   # 2) Check in the boot media (as with USB provisioning)
+   # 3) Check on the "big" partition (as with safemode->rauc migrations)
+
+   if [ -z "$(ls -A /payload 2>dev/null)" ] ; then
```

Comment:
You're missing the forward slash in 'dev/null' here and in the line below.

Reply:
Nice catch - both fixed!